common/flatpak-dir.c

```diff
@@ -13720,9 +13720,33 @@ local_match_prefix (FlatpakDir *self,
   return matches;
 }
 
+static gboolean
+repo_get_ref_collection_binding (OstreeRepo  *repo,
+                                 const char  *ref,
+                                 const char  *commit,
+                                 char       **out_collection_id,
+                                 GError     **error)
+{
+  g_autoptr(GVariant) commit_v = NULL;
+  g_autoptr(GVariant) commit_metadata = NULL;
+  const char *collection_id = NULL;
+
+  if (!ostree_repo_load_commit (repo, commit, &commit_v, NULL, error))
+    return FALSE;
+
+  commit_metadata = g_variant_get_child_value (commit_v, 0);
+  g_variant_lookup (commit_metadata, "ostree.collection-binding", "&s", &collection_id);
```

Comment:
Hmm, there was some work going on to support multiple collections bindings for a commit. Will that not break this?

Reply:
Yeah I think if https://github.com/ostreedev/ostree/pull/1805 is merged we'll have to give this some more thought and figure out what to do here and in the other places we use `ostree.collection-binding`. But as I said in the commit message, using the remote's collection ID isn't perfect either (it might not match that of the ref).

Reply:
I filed https://github.com/flatpak/flatpak/issues/3103 about this.